app/src/main/java/com/github/se/travelpouch/model/home/StorageDashboardViewModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.github.se.travelpouch.model.home
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlin.math.max
+import kotlin.math.min
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+fun formatStorageUnit(number: Long): String {
+  val units = arrayOf("B", "KiB", "MiB", "GiB")
+  var size = number.toDouble()
+  var unitIndex = 0
+
+  while (size >= 1024 && unitIndex < units.size - 1) {
+    size /= 1024
+    unitIndex++
+  }
+
+  return String.format(null, "%.1f %s", size, units[unitIndex])
```

Comment:
I wanted to use string interpolation because it is way easier to read but it's not as capable as `String.format` regarding the numeric formatting. You can find more information about [in the docs](https://kotlinlang.org/docs/strings.html#string-formatting)

Reply:
Sorry for the comment, you are right